lang/cs/Org.Apache.REEF.IO/PartitionedData/FileSystem/FilePartition.cs

```diff
@@ -0,0 +1,129 @@
+ * under the License.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Org.Apache.REEF.IO.FileSystem;
+using Org.Apache.REEF.IO.FileSystem.Hadoop;
+using Org.Apache.REEF.IO.FileSystem.Local;
+using Org.Apache.REEF.IO.PartitionedData.FileSystem.Parameters;
+using Org.Apache.REEF.IO.PartitionedData.Random.Parameters;
+using Org.Apache.REEF.Tang.Annotations;
+using Org.Apache.REEF.Utilities.Logging;
+
+namespace Org.Apache.REEF.IO.PartitionedData.FileSystem
+{
+    internal sealed class FilePartition<T> : IPartition<IEnumerable<T>>
+    {
+        private static readonly Logger Logger = Logger.GetLogger(typeof(FilePartition<T>));
+
+        private readonly string _id;
+        private readonly ISet<string> _filePaths;
+        private readonly IFileSystem _fileSystem;
+        private readonly IFileSerializer<T> _fileSerializer;
+        private IList<string> localFileNames = new List<string>();
+
+        [Inject]
+        private FilePartition([Parameter(typeof(PartitionId))] string id,
+            [Parameter(typeof(FilePathsInPartition))] ISet<string> filePaths,
+            IFileSystem fileSystem,
+            IFileSerializer<T> fileSerializer)
+        {
+            _id = id;
+            _filePaths = filePaths;
+            _fileSystem = fileSystem;
+            _fileSerializer = fileSerializer;
+
+            if (fileSystem is HadoopFileSystem)
+            {
+                Logger.Log(Level.Info, "+++++HadoopFileSystem");
+            }
+
+            if (fileSystem is LocalFileSystem)
+            {
+                Logger.Log(Level.Info, "!!!!!LocalFileSystem");
+            }
+        }
+
+        public string Id
+        {
+            get { return _id; }
+        }
+
+        /// <summary>
+        /// This method copy remote files to local and then deserialize the files.
+        /// It returns the IEnumerble of T, the details is defined in the Deserialize() method 
+        /// provided by the Serializer
+        /// </summary>
+        /// <returns></returns>
+        public IEnumerable<T> GetPartitionHandle()
+        {
+            foreach (var f in _filePaths)
+            {
+                Logger.Log(Level.Info, "remoteFileName: " + f);
+                localFileNames.Add(CopyFromRemote(f));
+            }
+
+            return _fileSerializer.Deserialize(localFileNames);
+        }
+
+        private string CopyFromRemote(string sourceFilePath)
+        {
+            Logger.Log(Level.Info, "GetUriPrefix: " + _fileSystem.UriPrefix);
+
+            Uri sourceUri = new Uri(_fileSystem.UriPrefix + sourceFilePath);
+            Logger.Log(Level.Info, "remoteUri: " + sourceUri);
+
+            if (!_fileSystem.Exists(sourceUri))
+            {
+                throw new ApplicationException("Remote File does not exists.");
+            }
+
+            var localFilePath = Path.GetTempPath() + "-partition-" + DateTime.Now.ToString("yyyyMMddHHmmssfff");
```

Comment:
`Path.GetTempPath()` doesn't work on many YARN clusters. Instead, we need to replicate the temp file provider pattern from the java side. Often, we will only have write access to the working directory of the container. At other times, we might have to use specific folders for this.